.github/workflows/build-system-test-react-native.yml

```
@@ -0,0 +1,197 @@
+# Description: this workflow runs build mega-apps that have all the connected components for React Native.
+
+name: Build System Test Canary / React Native
+
+permissions:
+  id-token: write # This is required for aws-actions/configure-aws-credentials
+
+on:
+  schedule:
+    - cron: '0 * * * *' # Run at the first minute of every hour
+
+jobs:
+  build:
+    runs-on: macos-13
+    environment: ci
+    strategy:
+      fail-fast: false
+      matrix:
+        framework-version: [latest]
+        build-tool: [cli]
+        build-tool-version: [latest]
+        pkg-manager: [npm]
+        language: [ts]
+        node-version: [18]
+        platform: [ios, android]
+        logfile: [test.log]
+
+    env:
+      MEGA_APP_NAME: rn${{ matrix.framework-version }}Cli${{ matrix.build-tool-version }}Node${{ matrix.node-version }}${{ matrix.language }}${{ matrix.platform }}
+      DEPENDENCIES: aws-amplify @aws-amplify/ui-react-native react-native-safe-area-context amazon-cognito-identity-js @react-native-community/netinfo @react-native-async-storage/async-storage react-native-get-random-values react-native-url-polyfill
+
+    steps:
+      - name: Checkout Amplify UI
+        uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0 https://github.com/actions/checkout/commit/24cb9080177205b6e8c946b17badbe402adc938f
+        with:
+          persist-credentials: false
+      - name: Setup Node.js ${{ matrix.node-version }} with ${{ matrix.pkg-manager }}
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0 https://github.com/actions/setup-node/commit/64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: ${{ matrix.pkg-manager }}
+        env:
+          SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2
+
+      - name: Restore CocoaPods cache
+        if: ${{ matrix.platform == 'ios' }}
+        id: restore-cocoapods-cache
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1 https://github.com/actions/cache/commit/88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8
+        with:
+          path: ./examples/react-native/ios/Pods
+          key: ${{ runner.os }}-cocoapods-${{ inputs.commit }}
+          restore-keys: |
+            pods-${{ hashFiles('examples/react-native/ios/Podfile.lock') }}
+            pods-
```

Think this can be 
```suggestion
          restore-keys: pods-${{ hashFiles('examples/react-native/ios/Podfile.lock') }}
```

since that file is guaranteed to exist.